exonum/src/runtime/mod.rs

```diff
@@ -12,6 +12,71 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! Transactions runtime.
+//!
+//! This module contains common building blocks for creating runtimes for the Exonum blockchain.
+//!
+//! Each runtime contains specific services to execute transactions, process events,
+//! provide user APIs, e.t.c. There is a unified dispatcher that redirects all the calls
+//! and requests to the appropriate runtime environment. Thus, blockchain interact with the
+//! dispatcher, and not with a specific runtime instance.
+//!
+//! # Service life cycle
+//!
+//! 1. Each runtime has its own [artifacts] registry from which users can deploy them. The artifact
+//! identifier is required by the runtime for constructing service instances. In other words,
+//! an artifact identifier means same as class name, and a specific service instance is
+//! the class instance.
+//!
+//! 2. Each validator administrator should request the dispatcher to deploy an artifact
+//! and then validator node should send confirmation if this request is successful. Then, if the
+//! number of confirmations is equal to the total number of validators, each validator calls the
+//! dispatcher to register the artifact as deployed. After that validators can send requests to
+//! start new services instances from this artifact.
+//!
+//! 3. To start a new service instance, each validator administrator should send request
+//! to dispatcher. Each request contains the artifact identifier, instance name, and
+//! instance configuration parameters. Then, as in the previous case, if the number of
```

Comment:
Is that expected that all of these parameters are the same in each request (i.e., each administrator must specify exactly the same parameters)?

Reply:
Yes, each administrator much specify exactly the same parameters.